server/src/main/java/edu/jhu/server/Game.java

```diff
@@ -52,17 +66,93 @@ public void notifyPlayers(JSONObject event) {
       player.sendEvent(event);
     }
   }
-
+  
+  private JSONObject makeChatMessage(String body) {
+	  JSONObject chat = new JSONObject();
+	  chat.put("eventType", "CHAT_NOTIFICATION");
+	  chat.put("author", "Game");
+	  chat.put("body", body);
+	  return chat;
+  }
+  
+  private JSONObject makeInvalidRequestMessage(String player, String reason){
+	  JSONObject invalidRequest = new JSONObject();
+	  invalidRequest.put("eventType", "INVALID_REQUEST_NOTIFICATION");
+	  invalidRequest.put("author", "Game");
+	  invalidRequest.put("reason", reason);
+	  invalidRequest.put("player", player);
+	  return invalidRequest;
+  }
+  
+  private JSONObject makeMoveNotification(Suspect suspect, ILocation location){
+	  JSONObject move = new JSONObject();
+	  move.put("eventType", "MOVE_NOTIFICATION");
+	  return move;
+  }
+  
+  
+  
+  private void provideEvidence(CaseFile casefile, Player suggester) {
+	  Player playerWithEvidence = null;
+	  for (Player player : players) {
+		  if(player.getTag() != suggester.getTag() && 
+			(player.hasCard(casefile.getRoom()) || 
+			player.hasCard(casefile.getSuspect()) ||
+			player.hasCard(casefile.getWeapon()))) {
+			  playerWithEvidence = player;
+			  break;
+		  }  
+	  }
+	  if (playerWithEvidence == null) {
+		  JSONObject chat = makeChatMessage("Nobody could provide evidence against this suggestion!");
+		  handleEvent(chat);
+	  }
+  }
+  
+  private void handleSuggestion(JSONObject accusation, Player suggester) {
+	  ILocation suggestedRoom = board.getLocationOf(suggester.getSuspect());
+	  Suspect theAccused = Suspect.get(accusation.get("suspect").toString());
+	  Weapon theWeapon = Weapon.get("knife");
+	  CaseFile casefile = new CaseFile((Room) suggestedRoom, theAccused, theWeapon);
+	  if (suggestedRoom instanceof Room) {
+			JSONObject chat = makeChatMessage(suggester.getTag() +
```

Comment:
my complaint about sending a chat message here is that the frontend has no way of differentiating this event from a regular chat event. so say if the frontend wanted to do something fancy when someone suggested something, it would have to parse all the chat messages.

i think that you should send a SuggestionNotification message, which contains the suspect/weapon/room, and the frontend can deal with turning it into a chat message.